overlord/servicestate/servicemgr.go

```diff
@@ -194,6 +194,8 @@ func delayedCrossMgrInit() {
 	// hook into conflict checks mechanisms
 	snapstate.AddAffectedSnapsByAttr("service-action", serviceControlAffectedSnaps)
 	snapstate.SnapServiceOptions = SnapServiceOptions
+	snapstate.AllQuotas = AllQuotas
+	snapstate.RemoveSnapFromQuota = RemoveSnapFromQuota
```

Comment:
it's a strange division of responsibility to have to export both of these to snapstate, I understand that RemovSnapFromQuota might be useful on its own, but for snapstate purposes we should from export something like MaybeRemoveSnapFromQuotas(st, snapName) that does what the code in snapstate does all in servicestate.

Reply:
done